nifi-nar-bundles/nifi-spring-bundle/nifi-spring-processors/src/main/java/org/apache/nifi/spring/SpringContextFactory.java

```diff
@@ -0,0 +1,139 @@
+                SpringContextFactory.class.getClassLoader());
+        try {
+            InputStream delegateStream = contextCl.getResourceAsStream(SC_DELEGATE_NAME.replace('.', '/') + ".class");
+            byte[] delegateBytes = IOUtils.toByteArray(delegateStream);
+            Class<?> clazz = contextCl.doDefineClass(SC_DELEGATE_NAME, delegateBytes, 0, delegateBytes.length);
+            Constructor<?> ctr = clazz.getDeclaredConstructor(String.class);
+            ctr.setAccessible(true);
+            SpringDataExchanger springDelegate = (SpringDataExchanger) ctr.newInstance(config);
+            if (logger.isInfoEnabled()) {
+                logger.info("Successfully instantiated Spring Application Context from '" + config + "'");
+            }
+            return springDelegate;
+        } catch (Exception e) {
+            try {
+                contextCl.close();
+            } catch (Exception e2) {
+                // ignore
+            }
+            throw new IllegalStateException("Failed to instantiate Spring Application Context. Config path: '" + config
+                    + "'; Classpath: " + Arrays.asList(urls), e);
+        }
+    }
+
+    /**
+     *
+     */
+    private static URL[] gatherAdditionalClassPathUrls(String path) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Adding additional resources from '" + path + "' to the classpath.");
+        }
+        File libraryDir = new File(path);
+        if (libraryDir.exists() && libraryDir.isDirectory()) {
+            String[] cpResourceNames = libraryDir.list();
+            try {
+                URLClassLoader thisCl = (URLClassLoader) SpringContextFactory.class.getClassLoader();
+                List<URL> urls = new ArrayList<>();
+                for (int i = 0; i < cpResourceNames.length; i++) {
+                    if (!isDuplicate(thisCl.getURLs(), cpResourceNames[i])) {
+                        URL url = new File(libraryDir, cpResourceNames[i]).toURI().toURL();
+                        urls.add(url);
+                        if (logger.isDebugEnabled()) {
+                            logger.debug("Identifying additional resource to the classpath: " + url);
+                        }
+                    }
+                }
+                return urls.toArray(new URL[] {});
+            } catch (Exception e) {
+                throw new IllegalStateException(
+                        "Failed to parse user libraries from '" + libraryDir.getAbsolutePath() + "'", e);
+            }
+        } else {
+            throw new IllegalArgumentException("Path '" + libraryDir.getAbsolutePath()
+                    + "' is not valid because it doesn't exist or does not point to a directory.");
+        }
+    }
+
+    /**
+     *
+     */
+    static boolean isDuplicate(URL[] currentURLs, String resourceName) {
+        if (resourceName.startsWith("spring")) {
+            resourceName = resourceName.substring(0, resourceName.lastIndexOf("-"));
```

Comment:
not sure I read the code flow correct, but if I have a jar in my application class path named springtony.jar, will this blow up?